### DISCUSSION TITLE
How to convert .shimple file to .jimple file?

### DISCUSSION THREAD

**Discussion by su-xingyu** [soot-oss/soot#2078] [Q&A] [answered]
Hi!

Does Soot provide a convenient way to convert a .shimple file to a .jimple file? I've checked the available command line options but it seems Soot doesn't accept .shimple files as input.

Thanks in advance!

**Comment by StevenArzt** [comment]
@MarcMil I know that someone from our team has been experimenting with Shimple lately (Florian maybe?) and I know that you discussed the matter with them. Can you provide any insights or refer to the colleague?

**Comment by MarcMil** [comment, marked as answer]
When using Soot as a library, the ShimpleBody has a ```toJimpleBody``` method.
https://github.com/soot-oss/soot/blob/7e7ac53f62541f8359e388bfc51e40359042854f/src/main/java/soot/shimple/ShimpleBody.java#L154

For the command line version:
When the wholeJimple option is set and the output format is set to Jimple, soot should remove the SSA form.
https://github.com/soot-oss/soot/blob/7e7ac53f62541f8359e388bfc51e40359042854f/src/main/java/soot/PackManager.java#L954

However, afaik, there right now there is no Shimple parser, so no way to get Shimple files into Soot.
